### PR TITLE
Top authors widget

### LINKED ISSUE #713: Add top authors widget
On the articles sidebar add a top article submitters widget similar to the "thanks to our community" widget on the forum overview.

https://laravel.io/articles

### PR DESCRIPTION
**Pull request by faissaloux**
resolves #713

### CODE REVIEW

**Comment by joedixon** [comment]
Hi @faissaloux,

Thanks for the contribution. Would you mind posting a screenshot of the update?

**Comment by faissaloux** [comment]
@joedixon

Screenshot from 2021-11-02 23-42-30

**Comment by driesvints** [comment]
Seems like the avatars don't align properly with the moderator widget. Can you check into making them align the same? On the forum overview page they align the same.

**Comment by faissaloux** [comment]
@driesvints fixed

Screenshot from 2021-11-03 11-52-54

File: app/Http/Controllers/Articles/ArticlesController.php [hunk 1]
```diff
@@ -33,11 +33,13 @@ public function index(Request $request)
             ->get();
         $moderators = User::moderators()->get();
         $canonical = canonical('articles', $request->only('sortBy', 'tag'));
+        $topSubmitters = User::mostSubmitsInLastDays(365)->take(5)->get();
```
Comment (driesvints) [inline app/Http/Controllers/Articles/ArticlesController.php:36]: Rename this to `$topAuthors` and use that terminology in the subsequent views as well.

File: resources/views/articles/index.blade.php [hunk 2]
```diff
@@ -21,6 +21,46 @@
                     @include('layouts._ads._forum_sidebar')
                 </div>
 
+                <div class="bg-white shadow rounded-md mt-6">
+                    <h3 class="text-xl font-semibold px-5 pt-5">
+                        Top article submitters
```
Comment (driesvints) [inline resources/views/articles/index.blade.php:26]:
```suggestion
                        Top authors
```

File: app/Http/Controllers/Articles/ArticlesController.php [hunk 3]
```diff
@@ -33,11 +33,13 @@ public function index(Request $request)
             ->get();
         $moderators = User::moderators()->get();
         $canonical = canonical('articles', $request->only('sortBy', 'tag'));
+        $topAuthors = User::mostSubmitsInLastDays(365)->take(5)->get();
```
Comment (joedixon) [inline app/Http/Controllers/Articles/ArticlesController.php:36]:
```suggestion
        $topAuthors = User::mostSubmissionsInLastDays(365)->take(5)->get();
```

File: app/Models/User.php [hunk 4]
```diff
@@ -249,11 +249,27 @@ public function scopeMostSolutions(Builder $query, int $inLastDays = null)
         }])->orderBy('solutions_count', 'desc');
     }
 
+    public function scopeMostSubmits(Builder $query, int $inLastDays = null)
```
Comment (joedixon) [inline app/Models/User.php:252]:
```suggestion
    public function scopeMostSubmissions(Builder $query, int $inLastDays = null)
```

File: app/Models/User.php [hunk 5]
```diff
@@ -249,11 +249,27 @@ public function scopeMostSolutions(Builder $query, int $inLastDays = null)
...
+            return $query;
+        }])->orderBy('articles_count', 'desc');
+    }
+
     public function scopeMostSolutionsInLastDays(Builder $query, int $days)
     {
         return $query->mostSolutions($days);
     }
 
+    public function scopeMostSubmitsInLastDays(Builder $query, int $days)
```
Comment (joedixon) [inline app/Models/User.php:268]:
```suggestion
    public function scopeMostSubmissionsInLastDays(Builder $query, int $days)
```

File: app/Models/User.php [hunk 6]
```diff
@@ -249,11 +249,27 @@ public function scopeMostSolutions(Builder $query, int $inLastDays = null)
...
+    {
+        return $query->mostSubmits($days);
```
Comment (joedixon) [inline app/Models/User.php:270]:
```suggestion
        return $query->mostSubmissions($days);
```

**Review by joedixon — CHANGES_REQUESTED** [review]
Just a small change to the scope name and then I think we're good to go. Thanks!

**Review by joedixon — APPROVED** [review]
This is awesome, thank you!

**Pull request merged**